File: src/main/java/com/github/codingvampyre/drachenfest/domain/Purchase.java
package com.github.codingvampyre.drachenfest.domain;

import lombok.*;
import org.hibernate.annotations.ColumnDefault;

import javax.persistence.*;
import java.util.List;

@Entity
@Getter
@Setter
@ToString
@RequiredArgsConstructor
@AllArgsConstructor
@Builder
public class Purchase extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "account_id")
    private Account account;

    @OneToMany(mappedBy = "purchase")
    private List<Ticket> tickets;

    private Float price;

    @ColumnDefault("false")
    private Boolean isPaid = false;

}

File: src/main/java/com/github/codingvampyre/drachenfest/domain/Purchase.kt
package com.github.codingvampyre.drachenfest.domain

import org.hibernate.annotations.ColumnDefault
import javax.persistence.*

@Entity
class Purchase : BaseEntity() {
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "account_id")
	private val account: Account? = null

	@OneToMany(mappedBy = "purchase")
	private val tickets: List<Ticket>? = null

	@Column(nullable = true)
	private val price: Float? = null

	@ColumnDefault("false")
	private val isPaid = false
}